src/asymmetric/builder.rs

```diff
@@ -0,0 +1,724 @@
+//! Type-safe asymmetric key generation.
+//!
+//! # Example
+//! ```
+//! use win_crypto_ng::asymmetric::{AsymmetricAlgorithm, AsymmetricAlgorithmId, AsymmetricKey};
+//! use win_crypto_ng::asymmetric::{ecc::NistP384, Ecdh, Rsa};
+//!
+//! // Handle to an asymmetric key whose algorithm ID is known at run-time
```

Comment:
I would add a textual explanation of the dynamic algorithm vs. the typed algorithm because it's not the most obvious thing to understand.

Reply:
Added a new 'Type-safety' section to `AsymmetricKey` that explains the reason/adds an example